src/vivarium_public_health/risks/distributions.py

```diff
@@ -1,403 +1,61 @@
-        return {'loc': pd.DataFrame(exposure['mean'], index=exposure.index),
-                'scale': pd.DataFrame(exposure['standard_deviation'], index=exposure.index)}
-
-
-class Weibull(BaseDistribution):
-
-    distribution = stats.weibull_min
-
-    def _get_params(self, exposure: pd.DataFrame) -> Dict[str, pd.DataFrame]:
-        def f(guess, mean, sd):
-            shape, scale = np.abs(guess)
-            mean_guess = scale * special.gamma(1 + 1 / shape)
-            var_guess = scale ** 2 * special.gamma(1 + 2 / shape) - mean_guess ** 2
-            return (mean - mean_guess) ** 2 + (sd ** 2 - var_guess) ** 2
-
-        opt_results = _get_optimization_result(exposure, f, lambda m, s: np.array((m, m / s)))
-        data_size = len(exposure)
-
-        if not np.all([opt_results[k].success is True for k in range(data_size)]):
-            raise NonConvergenceError('Weibull did not converge!!', 'weibull')
-
-        shape = np.abs([opt_results[k].x[0] for k in range(data_size)])
-        scale = np.abs([opt_results[k].x[1] for k in range(data_size)])
-        return {'c': pd.DataFrame(shape, index=exposure.index), 'scale': pd.DataFrame(scale, index=exposure.index)}
-
-
-class EnsembleDistribution:
-    """Represents an arbitrary distribution as a weighted sum of several concrete distribution types."""
-
-    def __init__(self, exposure, weights, distribution_map):
-        self.weights, self._distributions = self.get_valid_distributions(exposure, weights, distribution_map)
-
-    @staticmethod
-    def get_valid_distributions(exposure: pd.DataFrame, weights: pd.Series,
-                                distribution_map: Dict) -> Tuple[np.ndarray, Dict]:
-        """Produces a distribution that filters out non convergence errors and rescales weights appropriately.
-
-        Parameters
-        ----------
-        exposure :
-            Table where each row has a `mean` and `standard_deviation` for a single distribution.
-        weights :
-            A list of normalized distribution weights indexed by distribution type.
-        distribution_map :
-            Mapping between distribution name and distribution class.
-
-        Returns
-        -------
-            Rescaled weights and the subset of the distribution map corresponding to convergent distributions.
-        """
-        dist = dict()
-        for key in distribution_map:
-            try:
-                dist[key] = distribution_map[key](exposure)
-            except NonConvergenceError as e:
-                if weights[e.dist] > 0.05:
-                    weights = weights.drop(e.dist)
-                else:
-                    raise NonConvergenceError(f'Divergent {key} distribution has weights: {100*weights[key]}%', key)
+        builder.components.add_components(self._distributions.values())
 
-        return weights/np.sum(weights), dist
+    @classmethod
+    def get_distribution_map(cls):
```

Comment:
We don't want class methods for `get_distribution_map` (and therefore on `get_valid_distribution`).  With normal inheritance this method looks like:

```
def get_distribution_map(self):
    dist_map = super().get_distribution_map()
    return {dist_name: partial(SimulationDistribution, distribution=dist) for dist_name, dist in dist_map.items()}
```